src/libgosdt/include/configuration.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef CONFIGURATION_H
+#define CONFIGURATION_H
+
+#include <fstream>
+#include <string>
+
+#include "nlohmann/json.hpp"
+
+// Configuration object used to modify the algorithm behaviour
+// By design, all running instances of the algorithm within the same process
+// must share the same configuration
+class Configuration {
+    using json = nlohmann::json;
+
+   public:
+    Configuration() = default;
+
+    friend std::ostream &operator<<(std::ostream &ostream, const Configuration &config);
+
+    /// Constructs a new configuration object from a JSON object
+    static Configuration from_json(const json& object);
+
+    /// Dumps the configuration object to a JSON object
+    json to_json() const;
+
+    /// Save and Load the configuration object to a file
+    void save(const std::string &path) const;
+    static Configuration load(const std::string &path);
+
+   public:
+    // #TODO(Ilias): Replace the original configuration with these capitalized/renamed
```

Comment:
I'm ok leaving this as a todo for the merge, but if this is quickly resolvable now we may as well swap these in first.

Reply:
I'm leaving this in for now and will deal with it later!